Cesium.IntegrationTests/structs/union.c

```diff
@@ -0,0 +1,13 @@
+typedef struct {
+    union {
+        int i;
+        float f;
+    };
+} foo;
+
+int main() {
+    foo x;
+    x.f = 1.5f; // 1.5f as int == 1069547520 in Net 8.0. And OutOfMem in NetFramework
```

Comment:
Ok, how come it's `OutOfMemoryException`? Is it a problem of runtime or of our codegen?

Notably, this test seems to fail on Ubuntu as well (no .NET Framework), could you please take a look?

The exceptions seem to be
```
 [ERR] Unhandled exception. System.TypeLoadException: Size of field of type '<typedef>foo' from assembly 'out_cs, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' is too large.
 ```
 
 Perhaps we've miscalculated the size somehow?

Reply:
Currently, I believe that the reason for OOM is the same as for the demonstrated error.

If we figure out that the issue is in uninitialized structure (that would be very strange, though!), then feel free to open a separate issue and we may disable the test for the time being.

Reply:
Fixed it. Waiting for test results.